tests/interestTest.php

```diff
@@ -19,4 +20,109 @@ class interestsTests extends TestCase {
 	public function testNamespaceLoaded() {
 		$this->assertTrue( function_exists( '\\Pantheon\\EI\\WP\\Interest\\bootstrap' ) );
 	}
+
+	/**
+	 * Make sure get_interest exists.
+	 * @group wp-interest
+	 */
+	public function testGetInterestExists() {
+		$this->assertTrue(
+			function_exists( '\\Pantheon\\EI\\WP\\Interest\\get_interest' ),
+			'get_interest function does not exist'
+		);
+	}
+
+	/**
+	 * Make sure set_interest exists.
+	 * @group wp-interest
+	 */
+	public function testSetInterestExists() {
+		$this->assertTrue(
+			function_exists( '\\Pantheon\\EI\\WP\\Interest\\set_interest' ),
+			'set_interest function does not exist'
+		);
+	}
+
+	/**
+	 * Test the get_interest function.
+	 *
+	 * @dataProvider mockGetInterestData
+	 * @group wp-interest
+	 */
+	public function testGetInterest( array $interest_data ) {
+		// Get the actual data in a format that's easier to read.
+		$parsed_data = EI\HeaderData::parse( 'Interest', $interest_data );
+
+		$interest = Interest\get_interest( $interest_data );
+		$this->assertIsArray( $interest );
+		$this->assertNotEmpty(
+			$interest,
+			'Data is empty'
+		);
+		$this->assertEquals(
+			$interest,
+			$parsed_data,
+			'Data does not match'
+		);
+	}
+
+	/**
+	 * Test the set_interest function.
+	 * 
+	 * @group wp-interest
+	 */
+	public function testSetInterest() {
+		$input = [
+			'HTTP_IGNORED' => 'HTTP Ignored Entry',
+			'IGNORED_ENTRY' => 'Completely ignored entry',
+			'HTTP_SHOULD_BE_FOUND' => 'Should be found',
+			'HTTP_VARY' => 'Something, Wicked, This, Way',
+		];
+		$vary_header = EI\HeaderData::varyHeader( [ 'Comes' ], $input );
+
+		$interest = Interest\set_interest( [ 'Comes' ], $input );
+		$this->assertIsArray( $interest );
+		$this->assertNotEmpty(
+			$interest,
+			'Data is empty'
+		);
+		$this->assertEquals(
+			$interest,
+			$vary_header,
+			'Data does not match'
+		);
+	}
+
+	/**
+	 * Test the pantheon.ei.parsed_interest_data filter.
+	 *
+	 * @group wp-interest
+	 */
+	public function testParsedInterestData() {
+		$input = [ 'HTTP_INTEREST' =>'Carl Sagan|Richard Feynman|Albert Einstein' ];
+		// Filter the parsed interest data.
+		add_filter( 'pantheon.ei.parsed_interest_data', function( $interest_data ) {
+			return [ 'HTTP_INTEREST' =>'Carl Sagan|Richard Feynman|Albert Einstein' ];
+		}, 10, 1 );
+
+		$data = Interest\get_interest( $input );
+		$this->assertEquals(
+			$data,
+			$input,
+			'Parsed data does not match'
+		);
+	}
+
+	/**
+	 * Data provider for testGetInterests.
+	 *
+	 * @return array Mock interest data.
+	 */
+	private function mockGetInterestData() : array {
+		return [
+			[
+				'Carl Sagan|Richard Feynman|Neil deGrasse Tyson' => [ 'HTTP_INTEREST' => 'Carl Sagan|Richard Feynman|Neil deGrasse Tyson' ]
```

Comment:
The array key here doesn't need to echo the specific interests and, as such, is kinda long. This key is only shown when the test results fail so should probably be _descriptive_ but doesn't need to be exact. This is a minor point and no change is required, just an observation.